### PR TITLE
feat(rpc): trivial methods using new serialization trait

### PR DESCRIPTION
**Pull request by Mirko-von-Leipzig**
Implements some of the simpler RPC methods using the new trait.

Note that this explicly targets RPC v0.7-rc2 only. No backwards compatibility yet - let's fully support v0.7 and then we can work backwards. This let's us skip work for v0.4 and v0.5 if we manage to remove them before this work gets there.

------------------------

Particular review focus should be on the correctness of the DTO serialization ito inner types being used. i.e. ensure that if the spec says `BLOCK_HASH` is a `FELT` then the serialization implementation should use `serialize(&Felt(...))`. 

There isn't really a super easy way to enforce this apart from rigorous manual auditing when adding/updating DTOs.

------------------------

Adds DTO implementations for:
```
FELT
BLOCK_HASH
BLOCK_NUMBER
SYNC_STATUS
CHAIN_ID
```
and implements methods
```
starknet_blockHashAndNumber
starknet_blockNumber
starknet_chainId
starknet_syncing
```

Part of #1830

### CODE REVIEW

File: crates/rpc/src/dto/primitives.rs [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+        // Is the most significant nibble zero? We need to skip it if it is.
+        // Index is safe since we just checked that it is non-empty.
+        let zero_nibble = (0xF0 & data[0]) == 0;
+        to_hex_str(dbg!(data), dbg!(zero_nibble)).into()
+    }
+
+    pub fn bytes_to_hex_str_full(data: &[u8]) -> String {
+        to_hex_str(data, false)
+    }
```
Comment (kkovaacs) [inline crates/rpc/src/dto/primitives.rs:34]: Where do we use the full representation?

Comment (Mirko-von-Leipzig) [reply crates/rpc/src/dto/primitives.rs:34]: For Ethereum address: https://github.com/starkware-libs/starknet-specs/blob/master/api/starknet_api_openrpc.json#L1322-L1328

Comment (Mirko-von-Leipzig) [reply crates/rpc/src/dto/primitives.rs:34]: But not used in this PR

**Review by kkovaacs — APPROVED** [review]
LGTM, except for the `dbg!()` @sistemd already pointed out.

**Pull request merged**